### PR TITLE
feat: Add logging

### PR DESCRIPTION
**Pull request by diego-ojeda-binbash**
## Description
Add logging support by introducing 2 new variables:
* `tunnel1_log_options`
* `tunnel2_log_options`

## Motivation and Context
Logging is very useful for troubleshooting.

## Breaking Changes
There shouldn't be any as the new variables have safe defaults.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -71,6 +71,38 @@ resource "aws_vpn_connection" "default" {
...
 
+  dynamic "tunnel1_log_options" {
+    for_each = [var.tunnel1_log_options]
+
+    content {
+      dynamic "cloudwatch_log_options" {
+        for_each = tunnel1_log_options.value
+
+        content {
+          log_enabled       = cloudwatch_log_options.value.log_enabled
```
Comment (antonbabenko) [inline main.tf:82]: All of these `cloudwatch_log_options` options are optional, so you should use `lookup(cloudwatch_log_options.value, "log_enabled", null)`. Same in other optional arguments.

File: variables.tf [hunk 2]
```diff
@@ -277,6 +277,25 @@ variable "tunnel2_ike_versions" {
   default     = null
 }
 
+variable "tunnel1_log_options" {
+  description = "(Optional) Options for sending VPN tunnel logs to CloudWatch."
+  type = map(object({
```
Comment (antonbabenko) [inline variables.tf:282]: Please use `type = any`.

File: examples/complete-vpn-gateway-with-logging/main.tf [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  vpn_gateway_id      = module.vpc.vgw_id
+  customer_gateway_id = aws_customer_gateway.main.id
+
+  vpc_id                       = module.vpc.vpc_id
+  vpc_subnet_route_table_ids   = module.vpc.private_route_table_ids
+  vpc_subnet_route_table_count = length(var.vpc_private_subnets)
+  local_ipv4_network_cidr      = "0.0.0.0/0"
+  remote_ipv4_network_cidr     = module.vpc.vpc_cidr_block
+
+  tunnel1_log_options = {
```
Comment (antonbabenko) [inline examples/complete-vpn-gateway-with-logging/main.tf:17]: Please add these two tunnel log options to the existing complete example. There is no need to have a separate example just for this.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (diego-ojeda-binbash) [reply main.tf]: Done 👍

File: variables.tf (hunk 2 above)
Comment (diego-ojeda-binbash) [reply variables.tf]: Done 👍

File: examples/complete-vpn-gateway-with-logging/main.tf (hunk 3 above)
Comment (diego-ojeda-binbash) [reply examples/complete-vpn-gateway-with-logging/main.tf]: Done 👍

File: main.tf [hunk 4]
```diff
@@ -147,6 +179,38 @@ resource "aws_vpn_connection" "tunnel" {
...
```
Comment (antonbabenko) [inline main.tf:190]: Please apply the same fixes in all aws_vpn_connection resources.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: main.tf (hunk 4 above)
Comment (diego-ojeda-binbash) [reply main.tf]: Sorry about that. Done.

File: variables.tf [hunk 5]
```diff
@@ -277,6 +277,17 @@ variable "tunnel2_ike_versions" {
   default     = null
 }
 
+variable "tunnel1_log_options" {
+  description = "(Optional) Options for sending VPN tunnel logs to CloudWatch."
+  type        = any
+  default     = {}
+}
+variable "tunnel2_log_options" {
```
[the remark this answers is not in the document]
Comment (diego-ojeda-binbash) [reply variables.tf:286]: Done.

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
Thank you, everyone! :)

**Comment by nicon89** [comment]
In me-central-1 where logging is not enabled this is causing update on each apply:
```
      + tunnel1_log_options {
          + cloudwatch_log_options {
              + log_enabled       = (known after apply)
              + log_group_arn     = (known after apply)
              + log_output_format = (known after apply)
            }
        }

      + tunnel2_log_options {
          + cloudwatch_log_options {
              + log_enabled       = (known after apply)
              + log_group_arn     = (known after apply)
              + log_output_format = (known after apply)
            }
        }
```

Is there a way to disable it?

**Pull request merged**